File: tg-user-web/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller,userService){	
	
	$scope.entity={phone:'',password:'',username:''};
	
	//注册用户
	$scope.reg=function(){
		//比较两次输入的密码是否一致
		if($scope.password!=$scope.entity.password){
			alert("两次输入密码不一致，请重新输入");
			$scope.entity.password="";
			$scope.password="";
			return ;			
		}
		//新增
		userService.add($scope.entity,$scope.smscode).success(
			function(response){
				if(response.status) alert('注册成功');
				else alert(response.message);
			}		
		);
	}
    
	//发送验证码
	$scope.sendCode=function(){
		if($scope.entity.phone==null || $scope.entity.phone==""){
			alert("请填写手机号码");
			return ;
		}
		
		userService.sendCode($scope.entity.phone).success(
			function(response){
				if(response.status) alert("发送成功");
				else alert(response.message);
			}
		);		
	}
	//发送验证码
	$scope.showName=function(){
		userService.showName().success(
				function(response){
					$scope.loginName=response.loginName;
				}
		);		
	}
	
});	

File: tg-user-web/src/main/java/com/tg/fyc/user/controller/UserController.java
package com.tg.fyc.user.controller;

import java.util.HashMap;
import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.tg.fyc.common.PageResult;
import com.tg.fyc.common.PhoneFormatCheckUtils;
import com.tg.fyc.pojo.User;
import com.tg.fyc.user.api.UserApi;

@RestController
@RequestMapping("/user")
public class UserController {

	@Autowired
	private UserApi userApi;
	
	@RequestMapping("/register")
	public PageResult register(@RequestBody User user,String smscode){
		if(!userApi.checkSmsCode(user.getPhone(),smscode)) return PageResult.error(500,"验证码不正确");
		try {
			userApi.register(user);
			return PageResult.success(null);
		} catch (Exception e) {
			e.printStackTrace();
			return PageResult.error(500, "注册失败");
		}
	}
	
	@RequestMapping("/sendCode")
	public PageResult register(String phone){
		if(!PhoneFormatCheckUtils.isPhoneLegal(phone)){
			return PageResult.error(500, "手机号不存在");
		}
		try {
			userApi.createSmsCode(phone);
			return PageResult.success(null);
		} catch (Exception e) {
			e.printStackTrace();
			return PageResult.error(500, "信息发送失败");
		}
	}
	
	@RequestMapping("/showName")
	public Map showName(String phone){
		String name = SecurityContextHolder.getContext().getAuthentication().getName();
		Map map=new HashMap();
		map.put("loginName", name);
		return map;
	}
}
